### ISSUE TITLE
IntelliSense suggestions are not accessible to a screen reader

### ISSUE THREAD

**Issue by jayspadie** [Microsoft/monaco-editor#425]
monaco-editor version: 0.8.3
Browser: Any
OS: Windows 10

During Accessibility testing, it was found that IntelliSense suggestions are not read correctly by a screen reader.  Using Narrator with Edge, nothing is read while arrowing up/down the selections.  Using NVDA with Chrome or Firefox, the current line of code is repeated while arrowing through each suggestion instead of the suggestion itself, as shown below from the Monaco playground using NVDA with Chrome.

![image](https://cloud.githubusercontent.com/assets/5246869/25292825/efc0f4be-26a5-11e7-87f8-11fe9d534204.png)

**Comment by egamma** [comment]
// fyi @chrisdias

**Comment by chrisdias** [comment]
Related to https://github.com/Microsoft/vscode/issues/3787

**Comment by alexandrudima** [comment]
The root problem here is that in the monaco editor we don't have an aria-alert dom node configured. I am not sure if we are allowed to have multiple aria-alert dom nodes on a single page. i.e. what should happen if one host instantiates two monaco editors on a single page, should we create a single aria-alert dom node or should we create two. Should this be configurable such that we reuse the aria-alert dom node of the host...

**Comment by jayspadie** [comment]
It is not uncommon for us to have multiple aria-live="polite" (with aria-atomic="true") divs on a page, and they seem to work.  We also have cases of reusing the same aria-live div from multiple controls.  I am not aware of any restrictions with having more than one.

**Comment by alexandrudima** [comment]
@jayspadie In your experience, does `role="alert"` work on Narrator + Edge ? I could not find any way to make Narrator read an alert. (see #481)

e.g. from `https://www.w3.org/TR/WCAG20-TECHS/ARIA19.html`

```html
<!doctype html>
<html>
<head>
<meta charset="UTF-8">
<title>Using role=alert to Identify Errors</title>
<script
src="https://code.jquery.com/jquery-3.2.1.js"
integrity="sha256-DZAnKJ/6XZ9si04Hgrsxu/8s717jcIzLy3oi35EouyE="
crossorigin="anonymous"></script>

<script>
$(document).ready(function(e) {
	$('#signup').submit(function() {
		$('#errors').html('');
		if ($('#first').val() === '') {
			$('#errors').append('<p>Please enter your first name.</p>');
		}
		if ($('#last').val() === '') {
			$('#errors').append('<p>Please enter your last name.</p>');
		}
		if ($('#email').val() === '') {
			$('#errors').append('<p>Please enter your email address.</p>');
		}
		return false;
	});
	$('#signup2').submit(function() {
		$('#errors2').html('');
		if ($('#first2').val() === '') {
			$('#errors2').append('<p>Please enter your first name.</p>');
		}
		if ($('#last2').val() === '') {
			$('#errors2').append('<p>Please enter your last name.</p>');
		}
		if ($('#email2').val() === '') {
			$('#errors2').append('<p>Please enter your email address.</p>');
		}
		return false;
	});
	$('#signup3').submit(function() {
		$('#errors3').html('');
		if ($('#first3').val() === '') {
			$('#errors3').append('<p>Please enter your first name.</p>');
		}
		if ($('#last3').val() === '') {
			$('#errors3').append('<p>Please enter your last name.</p>');
		}
		if ($('#email3').val() === '') {
			$('#errors3').append('<p>Please enter your email address.</p>');
		}
		return false;
	});
});
</script>
</head>

<body>
<h1>Using ARIA Live Regions or role=alert to Identify Errors</h1>
<form name="signup" id="signup" method="post" action="">
  <p id="errors" role="alert" aria-atomic="true"></p>
  <p>
    <label for="first">First Name (required)</label><br>
    <input type="text" name="first" id="first">
  </p>
  <p>
    <label for="last">Last Name (required)</label><br>
    <input type="text" name="last" id="last">
  </p>
  <p>
    <label for="email">Email (required)</label><br>
    <input type="text" name="email" id="email">
  </p>
  <p>
    <input type="submit" name="button" id="button" value="Submit">
  </p>
</form>
</body>
</html>
```
